File: src/atcoder/abc116/src/C.java
import java.util.Arrays;
import java.util.Comparator;
import java.util.List;
import java.util.Scanner;
import java.util.stream.Collectors;

public class C {

    private static Scanner scanner = new Scanner(System.in);

    public static void main(String[] args) {
        int N = scanner.nextInt();
        int[] h = new int[N];
        for (int i = 0; i < N; i++) h[i] = scanner.nextInt();
        List<Integer> hList = Arrays.stream(h).boxed().sorted(Comparator.reverseOrder()).collect(Collectors.toList());

        int max = hList.get(0);

        int count = 0;
        int[] flowers = h;
        for (int i = 0; i < max; i++) {
            StringBuilder builder = new StringBuilder();
            for (int flower : flowers) builder.append(flower > 0 ? String.valueOf(flower).replace("0", "A") : flower);
            count += Arrays.stream(builder.toString().split("0")).filter(s -> !s.trim().equals("")).count();

            for (int j = 0; j < N; j++) {
                if (flowers[j] > 0) flowers[j]--;
            }
        }

        System.out.println(count);
    }

}

File: src/atcoder/abc116/src/C.kt
private val scanner = java.util.Scanner(System.`in`)

private fun readInts(n: Int) = Array(n) { scanner.nextInt() }
private fun readLongs(n: Int) = Array(n) { scanner.nextLong() }
private fun readLineInts() = readLine()!!.split(" ").map { it.toInt() }.toTypedArray()
private fun readLineLongs() = readLine()!!.split(" ").map { it.toLong() }.toTypedArray()

fun main(args: Array<String>) {
    val N = readLine()!!.toInt()
    val h = readInts(N).toList()

    var count = 0
    val flowers = h.toMutableList()
    for (w in 0 until h.max()!!) {
        val line = flowers.joinToString("") { if (it > 0) it.toString().replace("0", "A") else it.toString() }
        count += line.split("0").filter { it.trim() != "" }.size
        for ((index, value) in flowers.withIndex()) {
            if (value > 0) flowers[index]--
        }
    }

    println(count)
}